### PR TITLE
Add Golang Fuzz Task

### PR DESCRIPTION
**Pull request by jerop**
# Changes

This task is a Golang Tekton Task to fuzz Go projects

It uses the native Go Fuzzing Beta for dynamic analysis to find issues
in your code.

The release announcement https://blog.golang.org/fuzz-beta notes that:
"Fuzzing is a type of automated testing which continuously manipulates
inputs to a program to find issues such as panics or bugs. These
semi-random data mutations can discover new code coverage that existing
unit tests may miss, and uncover edge case bugs which would otherwise
go unnoticed. Since fuzzing can reach these edge cases, **fuzz testing is
particularly valuable for finding security exploits and vulnerabilities**"

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Follows the [authoring recommendations](https://github.com/tektoncd/catalog/blob/main/recommendations.md)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality of task changed or new task added)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [x] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [x] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [x] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [x] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [x] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [x] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md)
for more details._

---

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]:  https://github.com/tektoncd/catalog/issues/413

### CODE REVIEW

File: task/golang-fuzz/OWNERS [hunk 1]
```diff
@@ -0,0 +1,4 @@
+approvers:
+- jerop
+reviewers:
+- jerop
```
Comment (bobcatfish) [inline task/golang-fuzz/OWNERS:4]: you can add me too if you want! :D

**Review by bobcatfish — COMMENTED** [review]
/approve

File: task/golang-fuzz/OWNERS (hunk 1 above)
Comment (jerop) [reply task/golang-fuzz/OWNERS:5]: I WANT 😁

**Comment by chmouel** [comment]
/lgtm

Looking Good! 🤙🏽
